### ISSUE TITLE
Is `count` measured in wide characters or bytes?

### ISSUE THREAD

**Issue by ptaffet** [MicrosoftDocs/cpp-docs#1065]
Most of the documentation seems to indicate that `count` is measured in bytes, but the Important box in the remarks section seems to suggest it is the number of wide characters. Which is correct? The example does not make this clear.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 79027810-d629-5a02-4e23-58164db42d1e
* Version Independent ID: fb2f489f-343d-9967-7965-fbea40443d53
* Content: [wcstombs_s, _wcstombs_s_l](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/wcstombs-s-wcstombs-s-l?view=vs-2019#feedback)
* Content Source: [docs/c-runtime-library/reference/wcstombs-s-wcstombs-s-l.md](https://github.com/Microsoft/cpp-docs/blob/master/docs/c-runtime-library/reference/wcstombs-s-wcstombs-s-l.md)
* Product: **visual-cpp**
* Technology: **cpp-ucrt**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by corob-msft** [comment]
@ptaffet 

Yes, the note is confusing. I'm not sure what it was originally trying to convey, but I think the intent of the note is to make sure there are sufficient bytes in *count* to account for all the wide characters to convert.

The *count* parameter refers strictly to the number of bytes allowed in the converted multi-byte string before the terminating null. It must be at least one byte less than the buffer size, to account for the terminating null. (The example appears to have an error in this regard.) The value of *count* should also be sufficiently large to hold the converted string, or an ERANGE error results. Given the nature of traditional DBCS encodings, such as Shift-JIS, Big5, or GB2312, there are never more than twice as many bytes output as the number of wide characters input. However, not every wide character is representable in these encodings. This is not true of conversion to UTF-8, which can require 3 or 4 bytes to encode many characters that only require 2 bytes in the DBCS encodings, but which can encode every possible wide character. The upshot is, the heuristic for the number of bytes to allocate for the converted string depends on which MBCS character set you use.

**Comment by ptaffet** [comment]
Thanks. It would be nice if the documentation in particular clarified the difference between `sizeInBytes` and `count`, other than the -1 for the null terminator and the ability to pass in `_TRUNCATE`.

**Comment by TylerMSFT** [comment]
We are in the process of moving away from GitHub issues for C++ documentation (we still accept pull requests). For details, please see [Announcing a new way to give feedback on Microsoft Learn](https://techcommunity.microsoft.com/t5/microsoft-learn-blog/announcing-a-new-way-to-give-feedback-on-microsoft-learn/ba-p/4027635) This issue will be closed after it is copied to an internal tracking system.
